persistent-template/Database/Persist/TH.hs

```diff
@@ -1251,50 +1252,56 @@ maybeTyp :: Bool -> Type -> Type
 maybeTyp may typ | may = ConT ''Maybe `AppT` typ
                  | otherwise = typ
 
--- | produce code similar to the following:
+
+
+entityToPersistValueHelper :: (PersistEntity record) => record -> PersistValue
+entityToPersistValueHelper entity = PersistMap $ zip columnNames fieldsAsPersistValues
+    where
+        columnNames = map (unHaskellName . fieldHaskell) (entityFields (entityDef (Just entity)))
```

Comment:
I don't know why there is a `Monad` constraint on the `entityDef` function. It looks like the workaround for this elsewhere in the codebase is to wrap `entity` in a `Just`.

Reply:
It's an artifact of a time before `Proxy`, I believe - the idea is that you can write `Nothing :: Maybe Person` to signal the type, or `return undefined` or similar.